great_expectations/core/expectation_suite.py

```diff
@@ -174,8 +184,30 @@ def add_expectation(self, expectation: Expectation) -> Expectation:
                     raise exc
 
         expectation.register_save_callback(save_callback=self._save_expectation)
+
+        self._submit_expectation_created_event(expectation=expectation)
+
         return expectation
 
+    def _submit_expectation_created_event(self, expectation: Expectation) -> None:
+        if expectation.__module__.startswith("great_expectations."):
+            custom_exp_type = False
+            expectation_type = expectation.expectation_type
+        else:
+            custom_exp_type = True
+            anonymizer = Anonymizer()
+            expectation_type = anonymizer.anonymize(expectation.expectation_type)
+
+        submit_event(
+            event=ExpectationSuiteExpectationCreatedEvent(
+                action=EXPECTATION_SUITE_EXPECTATION_CREATED,
```

Comment:
It feels like a smell to me that we have a class just for this action, and we still have to pass the action in. IMO we should either have a generic ExpectationSuiteExpectationEvent that takes in the action, or the event should know its own action and instantiate itself with that.

Reply:
I agree but the shape of the data is slightly different between create vs delete/update